### ISSUE TITLE
Feature failures is calculated incorrectly

### ISSUE THREAD

**Issue by viastakhov** [renatoathaydes/spock-reports#173]
**Given** I have following test spec:
```java
import spock.lang.Specification
import spock.lang.Title
import spock.lang.Unroll

@Title("Testing Feature failures")
@Unroll
class MaxSpec extends Specification {

    def "Get the max value of two numbers"() {

        expect:
        verifyAll {
            Math.max(a, b) == c
            a + b == 4
        }

        where:
        a | b || c
        1 | 0 || 1
        2 | 3 || 3
        3 | 1 || 1
        4 | 2 || 4
        2 | 3 || 2
    }
}
```
**When** I execute this test
**Then** I got incorrect value of [Feature failures] in report
**Expected result**: 5
**Actual result**: 6

The issue causes due to using **verifyAll** wrapper which assert all statements inside its block

**Comment by renatoathaydes** [comment]
Looking at the Surefire report in a Maven build, it reports the same number as spock-reports:

```
-------------------------------------------------------------------------------
Test set: com.athaydes.MaxTest
-------------------------------------------------------------------------------
Tests run: 6, Failures: 6, Errors: 0, Skipped: 0, Time elapsed: 1.359 sec <<< FAILURE!
```

So this must be a "bug" in Spock itself, not in spock-reports. Please report to them.

**Comment by viastakhov** [comment]
Please kook at below you Surefire report, I've got following:
```
[INFO]
[INFO]
[ERROR] Tests run: 5, Failures: 5, Errors: 0, Skipped: 0
[INFO]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 10.685 s
[INFO] Finished at: 2019-04-28T12:56:06+03:00
[INFO] ------------------------------------------------------------------------
```
More over, when I run test by Intellij IDEA I also get 5 failed tests:
![Capture](https://user-images.githubusercontent.com/44951703/56862690-45f29180-69b6-11e9-8492-bcc2a32f7ca2.PNG)

So, it's not a Spock bug.

**Comment by renatoathaydes** [comment]
I tried it myself and Maven showed 6 tests, same as spock-reports. If you don't have the same results, please publish a project I can use to reproduce the problem.

**Comment by renatoathaydes** [comment]
Side question: why do you use `verifyAll`? It seems to me to be redundant as Spock always asserts all boolean conditions in the `then/expect` block, no?

**Comment by viastakhov** [comment]
> 
> 
> Side question: why do you use `verifyAll`? It seems to me to be redundant as Spock always asserts all boolean conditions in the `then/expect` block, no?

verifyAll wraps a assertion block in order to verify all test conditions at one test execution.

**Comment by viastakhov** [comment]
> 
> 
> I tried it myself and Maven showed 6 tests, same as spock-reports. If you don't have the same results, please publish a project I can use to reproduce the problem.

Here is the my project:
[sample.zip](https://github.com/renatoathaydes/spock-reports/files/3130524/sample.zip)

**Comment by renatoathaydes** [comment]
Thanks, I will try it tonight.

> verifyAll wraps an assertion block in order to verify all test conditions at one test execution.

I found a blog post explaining the difference: https://mrhaki.blogspot.com/2017/09/spocklight-group-assertions-with.html

Basically, `verifyAll` runs ALL assertions even if one of them fail, which is not the case without that. TIL.

**Comment by renatoathaydes** [comment]
I ran your sample project tests with DEBUG, here's the result:

```
[main] DEBUG com.athaydes.spockframework.report.SpecInfoListener - Before spec: MaxSpec
[main] DEBUG com.athaydes.spockframework.report.SpecInfoListener - Before feature: Get the max value of two numbers
[main] DEBUG com.athaydes.spockframework.report.SpecInfoListener - Before iteration: Get the max value of two numbers[0]
[main] DEBUG com.athaydes.spockframework.report.SpecInfoListener - Error on spec: MaxSpec
[main] DEBUG com.athaydes.spockframework.report.SpecInfoListener - After iteration: Get the max value of two numbers[0]
[main] DEBUG com.athaydes.spockframework.report.SpecInfoListener - Before iteration: Get the max value of two numbers[1]
[main] DEBUG com.athaydes.spockframework.report.SpecInfoListener - Error on spec: MaxSpec
[main] DEBUG com.athaydes.spockframework.report.SpecInfoListener - After iteration: Get the max value of two numbers[1]
[main] DEBUG com.athaydes.spockframework.report.SpecInfoListener - Before iteration: Get the max value of two numbers[2]
[main] DEBUG com.athaydes.spockframework.report.SpecInfoListener - Error on spec: MaxSpec
[main] DEBUG com.athaydes.spockframework.report.SpecInfoListener - After iteration: Get the max value of two numbers[2]
[main] DEBUG com.athaydes.spockframework.report.SpecInfoListener - Before iteration: Get the max value of two numbers[3]
[main] DEBUG com.athaydes.spockframework.report.SpecInfoListener - Error on spec: MaxSpec
[main] DEBUG com.athaydes.spockframework.report.SpecInfoListener - After iteration: Get the max value of two numbers[3]
[main] DEBUG com.athaydes.spockframework.report.SpecInfoListener - Before iteration: Get the max value of two numbers[4]
[main] DEBUG com.athaydes.spockframework.report.SpecInfoListener - Error on spec: MaxSpec
[main] DEBUG com.athaydes.spockframework.report.SpecInfoListener - Error on spec: MaxSpec
[main] DEBUG com.athaydes.spockframework.report.SpecInfoListener - After iteration: Get the max value of two numbers[4]
[main] DEBUG com.athaydes.spockframework.report.SpecInfoListener - After feature: Get the max value of two numbers
[main] DEBUG com.athaydes.spockframework.report.SpecInfoListener - After spec: MaxSpec
[main] DEBUG com.athaydes.spockframework.report.vivid.VividAstInspector - Trying to read source file /home/renato/programming/experiments/sample/src/test/groovy/MaxSpec.groovy
[main] DEBUG com.athaydes.spockframework.report.vivid.SpecSourceCodeCollector - Collecting code for class MaxSpec
[main] DEBUG com.athaydes.spockframework.report.vivid.SpecSourceCodeCollector - LABEL: expect -> verifyAll {
            Math.max(a, b) == c
            a + b == 4
        }
[main] DEBUG com.athaydes.spockframework.report.vivid.SpecSourceCodeCollector - LABEL: where -> a | b || c
[main] DEBUG com.athaydes.spockframework.report.internal.CssResource - Getting classpath resource text: spock-feature-report.css
[main] DEBUG com.athaydes.spockframework.report.internal.CssResource - Found css resource (2987 characters long)
[main] DEBUG com.athaydes.spockframework.report.internal.CssResource - Inlining css in HTML report
[ERROR] Tests run: 6, Failures: 6, Errors: 0, Skipped: 0, Time elapsed: 1.196 s <<< FAILURE! - in MaxSpec
```

As you can see, spock-reports is told about 6 test runs. Maven also shows 6 tests ran.

For some reason, at the end, after it shows the failures, it does report 5:

```
[ERROR] Tests run: 5, Failures: 5, Errors: 0, Skipped: 0
```

This is very strange.

It looks like Spock reports 5 iterations (notice there's 5 occurrences of `Before iteration`), but 6 failures (6 occurrences of `Error on spec: MaxSpec`). I can easily ignore the "stray" result at the end... so I will re-open the issue and fix it. But I still think it's a bug in Spock, not spock-reports :)

**Comment by renatoathaydes** [comment]
It's fixed now... will release the fixed code with the next version soon.
Sorry for closing the ticket too hastily before! I was initially unable to reproduce so assumed a bug elsewhere, but it turned out that multiple failures on a single iteration was actually counting towards total number of failures, which was wrong.

**Comment by renatoathaydes** [comment]
Fixed in version 1.6.3.
